src/app/shared/services/auth/providers/firebase.auth.ts

```diff
@@ -30,12 +31,51 @@ export class FirebaseAuthProvider extends AuthProviderBase {
     FirebaseAuthentication.addListener("authStateChange", (e) => {
       this.initialising$.next(true);
     });
-    // Ensure auth has settled by waiting for first emit.
-    // Emits `{user:null}` if not signed in, or user object if existing user re-authenticated
-    await firstValueFrom(this.initialising$.pipe(filter((v) => v === true)));
+
+    // Use different initialization strategies based on platform
+    if (Capacitor.getPlatform() === "ios") {
+      // iOS: Use polling method since authStateChange does not fire reliably
+      await this.waitForAuthInitialization();
+    } else {
+      // Non-iOS: Use authStateChange listener
+      await firstValueFrom(this.initialising$.pipe(filter((v) => v === true)));
+    }
     await this.handleAutomatedLogin();
   }
 
+  /**
+   * Wait for Firebase Auth to fully initialize and restore any existing session.
+   * This handles the race condition where getCurrentUser() might return null
+   * before Firebase Auth has finished restoring the session from persistent storage.
+   */
+  private async waitForAuthInitialization(): Promise<void> {
+    const maxAttempts = 6;
```

Comment:
nit(blocking): this feels like a somewhat arbitrary number. In practical testing did it ever take more than 1 attempt to settle on a user? Are there any other indicators we could use to (e.g. if we know there is a previous user stored in localStorage use that as an indication to retry)

Reply:
Yes, it is arbitrary. From my testing setup, it seemed to return populated user info on the third polling attempt. Good suggestion of exploring other indicators.